### ISSUE TITLE
running node doesn't work

### ISSUE THREAD

**Issue by mfreirehfexp** [dividab/tsconfig-paths#86]
Hi
I created this super simple repo to test if I could use this with node and couldn't make it work
https://github.com/mfreirehfexp/test-tsconfig-paths-node
I'm doing something wrong?

I tried to run this command after compiling
node -r tsconfig-paths/register main.js

Any help is appreciated, thanks!

**Comment by deskoh** [comment]
It's meant for running typescript source 'on-the-fly' without compiling. Typically used with `ts-node`.

**Comment by mfreirehfexp** [comment]
So why the docs have a node section?

**Comment by jonaskello** [comment]
@mfreirehfexp Yes it will work with node. The reason it does not work in your repo is that you are compiling to the `dist` folder and that folder does not have a `tsconfig.json`. So what you need to do is copy your tsconfig.json over to dist, make dist the current directory and then run `node -r tsconfig-paths/register main.js`. Now `tsconfig-paths` will find the `tsconfg.json` file and the paths in it. However since you have specified your paths as `src/some/common/files/dir` it will not work inside the `dist` folder (since it has no `src` folder). So in the `dist/tsconfig.json` you need to change the paths and remove the `src/` prefix:

```
    "paths": {
      "@shared/*": ["some/common/files/dir/*"],
      "@sharedv2": ["some/common/files/dir"]
    },
```

**Comment by jonaskello** [comment]
@mfreirehfexp  I made a pull request to your repo that fixes the problem.

**Comment by mfreirehfexp** [comment]
Thanks for the help! much appreciated

I suggest you may put it on the docs, something like:
> This runs compiling to the same folder, if your intention is compile to another folder you should include a copy of the tsconfig.json in it.

My original intention was to mimic nest.js project, in which they serve the app with ts-node but build it to production without it.

Thanks again for the help

**Comment by alexeychikk** [comment]
> @mfreirehfexp Yes it will work with node. The reason it does not work in your repo is that you are compiling to the `dist` folder and that folder does not have a `tsconfig.json`. So what you need to do is copy your tsconfig.json over to dist, make dist the current directory and then run `node -r tsconfig-paths/register main.js`. Now `tsconfig-paths` will find the `tsconfg.json` file and the paths in it. However since you have specified your paths as `src/some/common/files/dir` it will not work inside the `dist` folder (since it has no `src` folder). So in the `dist/tsconfig.json` you need to change the paths and remove the `src/` prefix:
> 
> ```
>     "paths": {
>       "@shared/*": ["some/common/files/dir/*"],
>       "@sharedv2": ["some/common/files/dir"]
>     },
> ```

Man, this must be in the docs. Otherwise it makes no sense mentioning in the readme that this package works for node.

**Comment by benwinding** [comment]
> Man, this must be in the docs. Otherwise it makes no sense mentioning in the readme that this package works for node.

I second this, using plain `node` require much more configuration. It seems to work out of the box with `ts-node` though...

**Comment by jonaskello** [comment]
A PR for the docs is welcome :-)

**Comment by jonathangaldino** [comment]
> @mfreirehfexp Yes it will work with node. The reason it does not work in your repo is that you are compiling to the `dist` folder and that folder does not have a `tsconfig.json`. So what you need to do is copy your tsconfig.json over to dist, make dist the current directory and then run `node -r tsconfig-paths/register main.js`. Now `tsconfig-paths` will find the `tsconfg.json` file and the paths in it. However since you have specified your paths as `src/some/common/files/dir` it will not work inside the `dist` folder (since it has no `src` folder). So in the `dist/tsconfig.json` you need to change the paths and remove the `src/` prefix:
> 
> ```
>     "paths": {
>       "@shared/*": ["some/common/files/dir/*"],
>       "@sharedv2": ["some/common/files/dir"]
>     },
> ```

Is it possible to reproduce the same results without having to copy tsconfig.json to the dist folder? I'm looking to create a bootstrap to register the correct paths, no luck yet :/

**Comment by moltar** [comment]
Just an FYI, aside from moving `tsconfig.json` to your build dir, you also need to change the `baseUrl` value, unless it is pointing to the root dir. E.g. if you had `./src` as value, then you'd want to change it to `./`.

**Comment by jonathangaldino** [comment]
> Just an FYI, aside from moving `tsconfig.json` to your build dir, you also need to change the `baseUrl` value, unless it is pointing to the root dir. E.g. if you had `./src` as value, then you'd want to change it to `./`.

Indeed. I forget to mention that, but a few minutes after my post I encounter the same problem.

**Comment by huypvw** [comment]
And it still not fix yet?
